src/pinch-zoom.js

```diff
@@ -587,6 +592,26 @@ var definePinchZoom = function () {
                 this.updatePlaned = false;
                 this.updateAspectRatio();
 
+                if (event && event.type === 'resize') {
+                    this.computeInitialOffset();
+                }
+
+                if (this.isInitialUpdate) {
+                    this.isInitialUpdate = false;
+                    this.computeInitialOffset();
+                    this.offset.x = this.initialOffset.x;
+                    this.offset.y = this.initialOffset.y;
+                    this.timeInitialised = Date.now();
+                }
```

Comment:
Instead of using `isInitialUpdate` plus the `2000 ms` load timer, how about something like this to detect if the image is loaded?

```js
function isImageLoaded(el) {
  return el.complete && el.naturalHeight !== 0;
}

// source: https://stackoverflow.com/a/34726863/118007
```

I'm thinking that it should be more stable than using a timer. What do you think?

Reply:
Sorry for the late reply, I moved to a different project and somehow lost track of this pull request.

Your suggestion seems fine to me. You should be able to commit to this branch as well.  :+1:

Reply:
@micgro42 No worries, thanks for replying! 😄 

I've made a few small adjustments, primarily to the centering commit (https://github.com/manuelstofer/pinchzoom/pull/52/commits/32dc2227c21691a35f546a7ea04ee1227ad727f1#diff-74f10427165b674d4d8a5c4bf321d992R102).

I added a image loading test in init, and used that to calculate the initial offset. It can also be fired via the 'load' event in the update method (just like you did it).

What do you think, does this make sense?

Just wanted to hear your thoughts before I merge!

Reply:
You made your changes with rebase, this made it a bit harder to review, but luckily I still had a local checkout of my changes so it wasn't too much of a problem 🙂

While I think your approach as mentioned above may work in principle, I'm not sure I completely understand how you implemented it.

https://github.com/cosmocode/ms-pinchzoom/blob/d8d26617a16fccac2563a1908f91169eda200a74/src/pinch-zoom.js#L241-L246

Shouldn't this either reference `this.el` or get `el` as a parameter? Also, does this work if `el` is not an image itself, but a `<div>`-wrapper around (multiple) images?

Reply:
@micgro42 Good points!

I've added a new commit, addressing these issues.

Reply:
Yeah, that looks good to me :+1: